src/main/kotlin/org/wycliffeassociates/resourcecontainer/ZipAccessor.kt

```diff
@@ -0,0 +1,92 @@
+package org.wycliffeassociates.resourcecontainer
+
+import java.io.File
+import java.io.IOException
+import java.io.BufferedReader
+import java.io.BufferedWriter
+import java.io.FileOutputStream
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+import java.util.zip.ZipOutputStream
+
+class ZipAccessor(private val file: File) : IResourceContainerAccessor {
+    private var _zipFile: ZipFile? = null
+
+    private fun openZipFile(): ZipFile {
+        if (_zipFile == null) _zipFile = ZipFile(file)
+        return _zipFile as ZipFile
+    }
+
+    private fun closeZipFile() {
+        _zipFile?.let(ZipFile::close)
+        _zipFile = null
+    }
+
+    @Synchronized
```

Comment:
is there a reason you're trying to think about thread safety here?

Reply:
Just because we really need the ZipFile to be closed in `write()`, and I don't want someone else triggering a reopen while the files are being deleted/renamed.

Reply:
Removed synchronization.